### PR TITLE
Applying all the feedback given from Code Review!

### PR DESCRIPTION
**Pull request by Mau-MD**
### PR Context
Since the main branch diverged a lot from the actual state of the app, it's gonna be complicated to merge all the PR's one by one. I propose the following:

I made this PR taking all the feedback given in previous PR's into account. I'm adding util functions, removing console logs, fixing tests, etc.

That way, if everything works out, this is the only PR's that needs to be merged into main and we won't have merge conflicts, because at the end, this PR contains everything from the older PR's.

Then we can close all the other PR's since they are already merged.

What do you think? @nosoyilse 

### This PR
- Add enums to several parts of the code. Redesigns a little bit the database schema so it's compatible with the new enum system
- Removes `documentation` resource kind, since it was never used in the app
- Fixes a bug where images were not displaying inside modals (this bug was made when migrating to enums)
- Adds two utils function that parses `limit`, `skip` and `query` from the request, to avoid writing duplicated code
- Fixes the cypress tests that were failing due to the guided tour appearing

### Next
Celebrate! 🎉

### IMPORTANT:
Since this PR was based in another one, a lot of commits appear here even if they are not related to this PR. In reaility commits start here: [making the use of typescript enums instead of plain numbers. to make …](https://github.com/Mau-MD/Learning-U/pull/71/commits/14d258dedb5e86438ff97fba534506c81c5f40fe)

### CODE REVIEW

File: backend/src/types/course.d.ts [hunk 1]
```diff
@@ -4,3 +4,8 @@ export interface ICourse extends DbObject {
   name: string;
   resources: any[];
 }
+
+export const enum CourseDifficulty {
```
Comment (nosoyilse) [inline backend/src/types/course.d.ts]: niceeeee! thanks for adding this

File: backend/src/utils/infiniteQuery.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { RequestWUser } from "../types/user";
+
+export const getInfiniteQueryParams = (req: RequestWUser) => {
+  const { limit, skip, query } = req.query;
+
+  if (
```
Comment (nosoyilse) [inline backend/src/utils/infiniteQuery.ts:6]: YESSSS!

**Review by nosoyilse — APPROVED** [review]
Amazing work overall Mauricio!

You were able to finish your goals and stretch goals.

I could see how you grew from the beginning to now, for example:
1. Commits at the beginning were super large and not very descriptive, it was very difficult to figure out what was going on, but the latest commit are small and easy to understand.
2. PRs didn't have any context nor test plan, it was very difficult to figure things out. You made a template and explain very clearly the objective of the PR and testing.
3. At the start you were writing very long functions, now code split into smaller chunks with very descriptive product names.

Plus you learnt unit testing, E2E testing, you integrated typescript, vercel and lint, plus all other react functionality. Besides all the ranking alg stuff, recommendation alg stuff and face some interesting problems like how to know courses where about the same and solved it very elegantly by using cosine similarity.

Since the beginning you were using component and interfaces properly keep doing that.

Some things that could improve:
- Don't commit console logs! This is a nit, because at the end of the PR you removed them, but still.
- Try to do the test in the same PR as the function, that way is easier to know what the test is testing and you will be doing test driven development. 
- Extract common code to util functions! it will save you time in the future.

I'm very happy to have been your manager at Meta and I'm looking forward to the things you will achieve in your future.

**Review by nosoyilse — APPROVED** [review]

**Comment by Mau-MD** [comment]
The pleasure is mine!!!

These 2 months have been the best 2 months in my entire life!!!! I've never had so much fun and the amount of knowledge I acquired in just a couple of weeks is magnificent. I did things that I would never image I could do before.

I learned a lot from you, and I'm already applying all these knowledge to personal and school projects!  I always tell my friends how amazing you are and I'm extremely glad you were my manager during this Internship. Thank you for having me, and hope to see you next year!

**Pull request merged**